src/promptflow/promptflow/_core/tool_meta_generator.py

```diff
@@ -310,6 +319,25 @@ def generate_tool_meta_dict_by_file(path: str, tool_type: ToolType):
         )
 
 
+def generate_flow_meta_dict_by_file(path: str, entry: str, source: str = None):
+    m = load_python_module_from_file(Path(path))
+    f = collect_flow_entry_in_module(m, entry)
+    tool = _parse_tool_from_function(f)
+
+    flow_meta = {"entry": entry, "function": f.__name__}
+    if source:
+        flow_meta["source"] = source
+    if tool.inputs:
+        flow_meta["inputs"] = {}
+        for k, v in tool.inputs.items():
+            flow_meta["inputs"][k] = {"type": [t.value for t in v.type]}
```

Comment:
I remember there is only one input type for a flow?

Reply:
Refined.